support-modules/acquisitions-stream/src/main/scala/com/gu/support/acquisitions/AcquisitionsStreamService.scala

```diff
@@ -0,0 +1,56 @@
+package com.gu.acquisition.services
+
+import java.nio.ByteBuffer
+
+import cats.data.EitherT
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.kinesis.AmazonKinesisAsyncClientBuilder
+import com.amazonaws.services.kinesis.model.{PutRecordRequest, PutRecordResult}
+import com.amazonaws.auth.AWSCredentialsProviderChain
+
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.control.NonFatal
+import com.gu.support.acquisitions.AcquisitionDataRow
+
+import io.circe.syntax._
+
+sealed trait AcquisitionsStreamServiceConfig {
+  val streamName: String
+}
+
+case class Ec2OrLocalConfig(val streamName: String, val credentialsProvider: AWSCredentialsProviderChain) extends AcquisitionsStreamServiceConfig
+case class LambdaConfig(val streamName: String) extends AcquisitionsStreamServiceConfig
+
+class AcquisitionsStreamService(config: AcquisitionsStreamServiceConfig, region: String = "eu-west-1") {
+
+  private val kinesisClient = {
+    val builder = AmazonKinesisAsyncClientBuilder.standard().withRegion(region)
+    config match {
+      case Ec2OrLocalConfig(_,provider) => builder.withCredentials(provider).build
+      case _: LambdaConfig => builder.build
+    }
+  }
+
+  def putAcquisition(acquisition: AcquisitionDataRow): EitherT[Future, String, Unit] = {
+    val promise = Promise[Either[String, Unit]]()
+
+    val request: PutRecordRequest = {
+      val buffer = ByteBuffer.wrap {
+        acquisition.asJson.noSpaces.getBytes
+      }
+
+      new PutRecordRequest()
+        .withStreamName(config.streamName)
+        .withPartitionKey(acquisition.identityId.getOrElse(acquisition.amount.toString))
```

Comment:
`amount` here is also an option, so I guess we shouldn't just use `.toString` on it here?

Reply:
Good spot.
The partition key is not especially important here (see [here](https://docs.aws.amazon.com/streams/latest/dev/key-concepts.html) for explanation) as the stream is low throughput.
We could just stringify the `eventTimeStamp` here instead?

Reply:
Yeah sure thing, so we're essentially going to be putting each event into its own shard?

Reply:
the stream's cloudformation actually defines a `ShardCount` of 1, so whatever we put here will make no difference! If there were more shards then the partition key would determine which one kinesis picks

Reply:
ahhh gotcha!